### DISCUSSION TITLE
When will “PhysicalFilter“ be used?

### DISCUSSION THREAD

**Discussion by caiwanli** [duckdb/duckdb#11453] [General]
When debugging the source code, executing the SQL statement 'select * from test where age>20 and age<25' does not trigger any methods in the PhysicalFilter. As far as I know, the PhysicalFilter should be the implementation of the WHERE clause.If the processing logic of PhysicalFilter is not executed, then which physical operator corresponds to 'where age>20 and age<25'?

**Comment by Tishj** [comment]
`src/execution/physical_plan/plan_filter.cpp`

Also there's a good chance it was optimized out, either disable the optimizer (`pragma disable_optimizer`) or use `EXPLAIN` (https://duckdb.org/docs/guides/meta/explain) to figure out what logical and physical operators are in your plan
